File: springboot-backend/src/main/java/com/capstone/project/service/PostService.java
package com.capstone.project.service;

import com.capstone.project.exception.ResourceNotFroundException;
import com.capstone.project.model.Class;
import com.capstone.project.model.Post;
import org.springframework.web.multipart.MultipartFile;

import java.util.List;
import java.util.Map;

public interface PostService {

    List<Post> getAllPost();

    List<Post> getAllPostByClassId(int id);

    Post getPostById(int id) throws ResourceNotFroundException;

//    Post createPost(Post post);

    Post createPost(Post post, List<String> files, int type);

    Post updatePost(Post posts, int id) throws ResourceNotFroundException;

    Boolean deletePost( int id) throws ResourceNotFroundException;

    Map<String, Object> getFilterPost(String search, String author, int classid, int page, int size) throws ResourceNotFroundException;
}

File: react-frontend/src/services/PostService.js
import axios from 'axios'

const API_BASE_URL = process.env.REACT_APP_HOST

const createPost = (post) => {
    return axios.post(API_BASE_URL + '/post', post, {
        headers: {
            Authorization: `Bearer ${localStorage.getItem('userToken')}`,
        },
    })
}

const updatePost = (id, post) => {
    return axios.put(API_BASE_URL + '/post/' + id, post, {
        headers: {
            Authorization: `Bearer ${localStorage.getItem('userToken')}`,
        },
    })
}

const deletePost = (id) => {
    return axios.delete(API_BASE_URL + '/post/' + id, {
        headers: {
            Authorization: `Bearer ${localStorage.getItem('userToken')}`,
        },
    })
}

const getAllPostByClassId = (id) => {
    return axios.get(API_BASE_URL + '/postbyclassid/' + id)
}

const getPostById = (id) => {
    return axios.get(API_BASE_URL + '/post/' + id)
}

const getFilterList = (
    search,
    author,
    fromcreated,
    tocreated,
    sortby,
    direction,
    classId,
    page,
    size
) => {
    return axios.get(
        API_BASE_URL +
            '/filterpost?search' +
            search +
            '&author' +
            author +
            '&fromcreated' +
            fromcreated +
            '&tocreated' +
            tocreated +
            '&sortby' +
            sortby +
            '&direction' +
            direction +
            '&classid' +
            classId +
            '&page' +
            page +
            '&size' +
            size
    )
}

const PostService = {
    createPost,
    getAllPostByClassId,
    updatePost,
    deletePost,
    getFilterList,
    getPostById,
}

export default PostService
